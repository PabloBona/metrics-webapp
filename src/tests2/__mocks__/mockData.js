const mockData = {
  coins: [
    {
      id: 'bitcoin',
      name: 'Bitcoin',
      rank: 1,
      symbol: 'BTC',
      price: 6362.74960614,
    },
    {
      id: 'ethereum',
      name: 'Ethereum',
      rank: 2,
      symbol: 'ETH',
      price: 208.130215868,
    },
    {
      id: 'ripple',
      name: 'XRP',
      rank: 3,
      symbol: 'XRP',
      price: 0.5227380884,
    },
    {
      id: 'bitcoin-cash',
      name: 'Bitcoin Cash',
      rank: 4,
      symbol: 'BCH',
      price: 507.88337011,
    },
    {
      id: 'stellar',
      name: 'Stellar',
      rank: 5,
      symbol: 'XLM',
      price: 0.2648749605,
    },
    {
      id: 'eos',
      name: 'EOS',
      rank: 6,
      symbol: 'EOS',
      price: 5.3374942604,
    },
    {
      id: 'litecoin',
      name: 'Litecoin',
      rank: 7,
      symbol: 'LTC',
      price: 50.4143886779,
    },
    {
      id: 'cardano',
      name: 'Cardano',
      rank: 8,
      symbol: 'ADA',
      price: 0.0750106925,
    },
    {
      id: 'monero',
      name: 'Monero',
      rank: 9,
      symbol: 'XMR',
      price: 104.697519052,
    },
    {
      id: 'tether',
      name: 'Tether',
      rank: 10,
      symbol: 'USDT',
      price: 0.9860800679,
    },
  ],
};

export default mockData;
