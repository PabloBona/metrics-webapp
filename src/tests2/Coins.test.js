import { render, screen } from '@testing-library/react';
import mockData from './__mocks__/mockData';

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  useSelector: jest.fn((selector) => selector({ coinSlice: { coin: mockData } })),
}));
jest.mock('../redux/slices/coinSlice', () => ({
  fetchCoin: jest.fn(() => Promise.resolve(mockData)),
}));

describe('Coins Component', () => {
  test('renders loading when data is not yet available', () => {
    render(
      <div className="col">
        <h2
          data-testid="h2test"
          className="align-items-center d-flex justify-content-center text-light vh-100"
        >
          Loading...
        </h2>
      </div>,
    );

    const loadingElement = screen.getByTestId('h2test');
    expect(loadingElement).toHaveTextContent('Loading...');
    expect(loadingElement).toMatchSnapshot();
  });

  test('renders Crypto Currency Stats when Render  ', () => {
    render(
      <h2 className="text-white bg-stats m-0">Crypto Currency Stats</h2>,
    );

    const loadingElement = screen.getByText('Crypto Currency Stats');
    expect(loadingElement).toHaveTextContent('Crypto Currency Stats');
    expect(loadingElement).toMatchSnapshot();
  });
});
