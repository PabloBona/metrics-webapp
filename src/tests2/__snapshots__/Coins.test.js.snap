// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coins Component renders Crypto Currency Stats when Render   1`] = `
<h2
  class="text-white bg-stats m-0"
>
  Crypto Currency Stats
</h2>
`;

exports[`Coins Component renders loading when data is not yet available 1`] = `
<h2
  class="align-items-center d-flex justify-content-center text-light vh-100"
  data-testid="h2test"
>
  Loading...
</h2>
`;
